src/utils/useStickyTabs.ts

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Tab } from '@terraware/web-components';
+
+import useQuery from './useQuery';
+import useStateLocation, { getLocation } from './useStateLocation';
+
+interface StickyTabsProps {
+  defaultTab: string;
+  tabs: Tab[];
+  viewIdentifier: string;
+}
+
+const makeTabSessionKey = (viewIdentifier: string) => `tab-${viewIdentifier}`;
```

Comment:
I went with local storage because that's what was decided when we wanted to make table filters sticky. I'm open to putting it wherever though. It seems like it is transient enough to not persist in a database but if that is the pattern we're already using I can move to that.

Reply:
Err... sorry. I meant session storage for the sticky filters. It was originally meant to go away if the user logs out.

Reply:
I'll ask in product-development to see where this should be stored.

Reply:
Ok, my understanding is that session storage is valid for the tab, local storage for the browser. Neither would be keyed off our login sessions - which spans tabs.

Reply:
Session storage last as long as the session isn't cleared. Generally session storage is cleared when you log out (but I am not sure exactly how we have implemented it), so it might just last as long as the browser is open. It is cross-browser-tab though

Reply:
Local storage just lasts between browser shut downs, that's the big difference.

Reply:
We don’t have our login sessions tied to session storage.. i can test using the console. Correct on the local storage. Have always used session storage for the page/tab view and not across tabs. Docs say something similar. Can confirm with a console test

Reply:
Yea, you're right. It is not cross tab, not sure where I got that idea but I've been thinking that a long time 😞

Reply:
So it sounds like either local storage or user preferences are the way to go

Reply:
There is a concept of BE session storage that lasts for the duration of user logged in . Not too familiar with that. Local storage sounds good but has a shared browser issue. User prefs would be user specific. Maybe local storage for simplicity and revisit later?

Reply:
I've got an open question in product-development, should be able to make the change quickly when they decide.

Reply:
Sounds good.

Reply:
Léann said session storage... I'll make sure to get another opinion on that but will merge this in for now, easy to change later.